osu.Game/Overlays/Music/PlaylistList.cs

```diff
@@ -76,12 +81,15 @@ public PlaylistList()
         public void AddBeatmapSet(BeatmapSetInfo beatmapSet)
         {
             items.Add(new PlaylistItem(beatmapSet) { OnSelect = itemSelected });
+            items.TriggerBeatmapSetsChange();
         }
 
         public void RemoveBeatmapSet(BeatmapSetInfo beatmapSet)
         {
-            PlaylistItem itemToRemove = items.Children.FirstOrDefault(item => item.BeatmapSetInfo == beatmapSet);
-            if (itemToRemove != null) items.Remove(itemToRemove);
+            PlaylistItem itemToRemove = items.Children.FirstOrDefault(item => item.BeatmapSetInfo.Hash == beatmapSet.Hash);
+            if (itemToRemove != null)
+                items.Remove(itemToRemove);
+            items.TriggerBeatmapSetsChange();
         }
 
         private class ItemSearchContainer : FillFlowContainer<PlaylistItem>, IHasFilterableChildren

```

Comment:
i'd rather this was at a framework level as a  `ReorderableFillFlowContainer<T> : FillFlowContainer<T> where T : IDraggable` or something